src/lib/builder.ts

```diff
@@ -1,11 +1,18 @@
-import isPromise from 'is-promise'
-
 import { BuilderHandler, Handler, HandlerCallback } from '../function/handler.js'
 import { HandlerResponse, BuilderResponse } from '../function/handler_response.js'
 import { HandlerContext, HandlerEvent } from '../function/index.js'
 
 import { BUILDER_FUNCTIONS_FLAG, HTTP_STATUS_METHOD_NOT_ALLOWED, METADATA_VERSION } from './consts.js'
 
+// stolen from https://github.com/then/is-promise/blob/master/index.mjs
+const isPromise = (obj: unknown): obj is Promise<unknown> =>
```

Comment:
```suggestion
const isPromise = <T = any>(obj: T | Promise<T>): obj is Promise<T>
```
This should work, and preserve the type

Reply:
actually, i've reverted this. we don't really know what type  is in the object, because it's returned by the user. also, we don't really need to know about it. `unknown` is fine here.